sdk/objectanchors/Azure.MixedReality.ObjectAnchors/src/ObjectAnchorsClient.cs

```diff
@@ -0,0 +1,218 @@
+            : this(accountId, accountDomain, new StaticAccessTokenCredential(token), options) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectAnchorsClient" /> class.
+        /// </summary>
+        /// <param name="accountId">The Azure Object Anchors account ID.</param>
+        /// <param name="accountDomain">The Azure Object Anchors account domain.</param>
+        /// <param name="keyCredential">The Azure Object Anchors account primary or secondary key credential.</param>
+        public ObjectAnchorsClient(Guid accountId, string accountDomain, AzureKeyCredential keyCredential)
+            : this(accountId, accountDomain, new MixedRealityAccountKeyCredential(accountId, keyCredential), new ObjectAnchorsClientOptions()) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectAnchorsClient" /> class.
+        /// </summary>
+        /// <param name="accountId">The Azure Object Anchors account ID.</param>
+        /// <param name="accountDomain">The Azure Object Anchors account domain.</param>
+        /// <param name="keyCredential">The Azure Object Anchors account primary or secondary key credential.</param>
+        /// <param name="options">The options.</param>
+        public ObjectAnchorsClient(Guid accountId, string accountDomain, AzureKeyCredential keyCredential, ObjectAnchorsClientOptions options)
+            : this(accountId, accountDomain, new MixedRealityAccountKeyCredential(accountId, keyCredential), options) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectAnchorsClient" /> class.
+        /// </summary>
+        /// <param name="accountId">The Azure Object Anchors account ID.</param>
+        /// <param name="accountDomain">The Azure Object Anchors account domain.</param>
+        /// <param name="credential">The credential used to access the Mixed Reality service.</param>
+        public ObjectAnchorsClient(Guid accountId, string accountDomain, TokenCredential credential)
+         : this(accountId, accountDomain, credential, new ObjectAnchorsClientOptions()) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectAnchorsClient" /> class.
+        /// </summary>
+        /// <param name="accountId">The Azure Object Anchors account ID.</param>
+        /// <param name="accountDomain">The Azure Object Anchors account domain.</param>
+        /// <param name="credential">The credential used to access the Mixed Reality service.</param>
+        /// <param name="options">The options.</param>
+        public ObjectAnchorsClient(Guid accountId, string accountDomain, TokenCredential credential, ObjectAnchorsClientOptions options)
+        {
+            Argument.AssertNotNull(accountDomain, nameof(accountDomain));
+            Argument.AssertNotNull(credential, nameof(credential));
+
```

Comment:
```suggestion
        /// <summary>
        /// Initializes a new instance of the <see cref="ObjectAnchorsClient" /> class.
        /// </summary>
        /// <param name="accountId">The Azure Object Anchors account ID.</param>
        /// <param name="accountDomain">The Azure Object Anchors account domain.</param>
        /// <param name="credential">The credential used to access the Mixed Reality service.</param>
        /// <param name="options">The options.</param>
        public ObjectAnchorsClient(Guid accountId, string accountDomain, TokenCredential credential, ObjectAnchorsClientOptions options = null)
        {
            Argument.AssertNotNull(accountDomain, nameof(accountDomain));
            Argument.AssertNotNull(credential, nameof(credential));

            options ??= new ObjectAnchorsClientOptions();
```

Reply:
See remarks at 33:55 [here](https://msit.microsoftstream.com/video/f603a4ff-0400-b564-fe5b-f1eb6b2e4cd0?referrer=https:%2F%2Fstatics.teams.cdn.office.net%2F). This doesn't have to apply to each constructor though, so we can discuss a single constructor to apply this logic to if we think that's better, but I'd raise an eyebrow at constructor consistency in that case.

Reply:
Reduced to a single constructor with optional options